lib/stdlib/src/gen_event.erl

```diff
@@ -263,81 +264,83 @@ send(M, Cmd) ->
     M ! Cmd,
     ok.
 
-loop(Parent, ServerName, MSL, Debug, true) ->
-     proc_lib:hibernate(?MODULE, wake_hib, [Parent, ServerName, MSL, Debug]);
-loop(Parent, ServerName, MSL, Debug, _) ->
-    fetch_msg(Parent, ServerName, MSL, Debug, false).
+loop(Parent, ServerName, MSL, AutoHibernateTimeout, Debug, true) ->
+     proc_lib:hibernate(?MODULE, wake_hib, [Parent, ServerName, MSL, AutoHibernateTimeout, Debug]);
+loop(Parent, ServerName, MSL, AutoHibernateTimeout, Debug, _) ->
+    fetch_msg(Parent, ServerName, MSL, AutoHibernateTimeout, Debug, false).
 
-wake_hib(Parent, ServerName, MSL, Debug) ->
-    fetch_msg(Parent, ServerName, MSL, Debug, true).
+wake_hib(Parent, ServerName, MSL, AutoHibernateTimeout, Debug) ->
+    fetch_msg(Parent, ServerName, MSL, AutoHibernateTimeout, Debug, true).
 
-fetch_msg(Parent, ServerName, MSL, Debug, Hib) ->
+fetch_msg(Parent, ServerName, MSL, AutoHibernateTimeout, Debug, Hib) ->
     receive
 	{system, From, Req} ->
 	    sys:handle_system_msg(Req, From, Parent, ?MODULE, Debug,
-				  [ServerName, MSL, Hib],Hib);
+				  [ServerName, MSL, AutoHibernateTimeout, Hib],Hib);
 	{'EXIT', Parent, Reason} ->
 	    terminate_server(Reason, Parent, MSL, ServerName);
 	Msg when Debug =:= [] ->
-	    handle_msg(Msg, Parent, ServerName, MSL, []);
+	    handle_msg(Msg, Parent, ServerName, MSL, AutoHibernateTimeout, []);
 	Msg ->
 	    Debug1 = sys:handle_debug(Debug, fun print_event/3,
 				      ServerName, {in, Msg}),
-	    handle_msg(Msg, Parent, ServerName, MSL, Debug1)
+	    handle_msg(Msg, Parent, ServerName, MSL, AutoHibernateTimeout, Debug1)
+    after AutoHibernateTimeout ->
+	    proc_lib:hibernate(?MODULE, wake_hib, [Parent, ServerName, MSL, AutoHibernateTimeout, Debug])
```

Comment:
Why not:
``` erlang
    loop(Parent, ServerName, MSL, AutoHibernateTimeout, Debug, true)
```
instead?  That would confine the `proc_lib:hibernate/3` call to one place.